client/src/components/graph/graph.tsx

```diff
@@ -727,33 +790,37 @@ class Graph extends React.Component<GraphProps, GraphState> {
     }
   }
 
-  mapScreenToPoint(pin: [number, number]) {
+  mapScreenToPoint(pin: [number, number]): vec2 {
     /*
         Map an XY coordinates from screen domain to cell/point range,
         accounting for current pan/zoom camera.
         */
     const { camera, projectionTF, modelInvTF, viewport } = this.state;
-    const cameraInvTF = camera!.invView();
+    const cameraInvTF = camera ? camera.invView() : null;
```

Comment:
I think the removal of this non-null assertions is making the code more complex that it needs to be. We can't get to any of these function calls without the state being defined

Reply:
I marked this single line, but it's a widespread pattern on this file. It might be worth conversation with @tihuan as well though

Reply:
After some further consideration - I'm under the opinion that we should stick with the conventional pattern and only use the `!` operator as last resort - specifically for this file as we are trying to reduce the amount of noise throughout the code and clean it up. 

I've reverted the last change made to this line to use the `?` operator instead.

Happy to open this up for a wider discussion - not sure this should be gating issue to merging

Reply:
oh wait, we can use optional chaining too? `camera?.invView();`